src/lib/mina_base/account.ml

```diff
@@ -528,28 +528,32 @@ module Checked = struct
 
   let amount_upper_bound = Bignum_bigint.(one lsl Amount.length_in_bits)
 
-  let min_balance_at_slot ~global_slot ~cliff_time ~cliff_amount ~vesting_period
+  let min_balance_at_slot ~global_slot ~cliff_time ~cliff_amount
+      ~(vesting_period : Mina_numbers.Global_slot_span.Checked.var)
       ~vesting_increment ~initial_minimum_balance =
-    let%bind before_cliff = Global_slot.Checked.(global_slot < cliff_time) in
+    let%bind before_cliff =
+      Global_slot_since_genesis.Checked.(global_slot < cliff_time)
+    in
     let%bind else_branch =
       make_checked (fun () ->
-          let _, slot_diff =
+          let _, (slot_diff : Mina_numbers.Global_slot_span.Checked.var) =
             Tick.Run.run_checked
-              (Global_slot.Checked.sub_or_zero global_slot cliff_time)
+              (Global_slot_since_genesis.Checked.diff_or_zero global_slot
+                 cliff_time )
```

Comment:
[check] Is `diff_or_zero` the same as `sub_or_zero`?

Reply:
`sub_or_zero` returns a `Global_slot_since_genesis.Checked.t`, while `diff_or_zero` returns a `Global_slot_span.Checked.t`.

For slots, `sub_or_zero` isn't so useful. I'd have to do some contortions to remove it from the interface.